server/render.js

```diff
@@ -82,11 +82,7 @@ async function doRender (req, res, pathname, query, {
   if (isResSent(res)) return
 
   const renderPage = (enhancer = Page => Page) => {
```

Comment:
I wonder if we could make this work:

```
// Default kept for backwards compatibility
const renderPage = (options = Page => Page) => {
  let EnhancedApp = App
  let EnhancedComponent = Component
  if(typeof options === 'function') {
    EnhancedComponent = options(Component)
  }
  if(typeof options === 'object') {
	EnhancedApp = options.enhanceApp(App)
    EnhancedComponent = options.enhanceComponent(Component)
  }
  const app = createElement(enhancer(EnhancedApp), { Component: EnhancedComponent, router, ...props })
}
```

Reply:
Usage is:

```
renderPage({
  enhanceApp: (App) => App,
  enhancePage: (Page) => Page
})
```

Reply:
@Tapppi @oliviertassinari what do you think?

Reply:
Forgot to mention that this would make it backwards compatible with the old implementation 👌

Reply:
Sounds like a great tradeoff.